Clinical-Genomics/microSALT
# Accreditation changes to report

# Description
Typing and QC report(s) have been updated. 
This primarily concern end-users. 
Much of the changes stemmed from the latest SWEDAC review.

End-user:
- Swedish translation
- Clarifications about the analysis used and limitations
- Reports are now given distinct versions (as opposed to implied by date)
- Drastic overhaul of the first four pages of the typing report (dates, method, verified organism)
- QC thresholds are now defined in config and auto-colors the QC report based on thresholds (NTC samples are excluded)
- Zebra table pattern, and minor field additions (like % reads compared to requested)
- Various other minor SWEDAC requirements (SWEDAC logos, Report generated line, end of report line)

Bioinformatics:
- Reports are now saved in an additional 'reports' folder for easier access
- microSALT now warns if fastq file exceeds 1GB
- Bug that generated a typing report despite qc_only is now resolved

Internals:
- microSALT now falls back to external sample name, if said sample doesn't resolve to a Clinical Genomics name
- Span criteria in the config is now percentage instead of decimal based, for consistency
- Bug that gave the database user-write permission is now fixed

## Primary function of PR


# Testing
On hasta:
- `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-microsalt-stage.sh BRANCHNAME`
- `us`
- `export MICROSALT_CONFIG=/home/proj/dropbox/microSALT.json`
- `source activate S_microSALT`
- `microSALT analyse project MIC3109 --qc_only`
- `microSALT analyse project MIC4107`
- `microSALT analyse project MIC4109`

1. Review the QC report (for MIC3109) and the Typing reports (for MIC4107 and MIC4109) to make sure the results do not deviate from those presented in AMSystem doc 1490 section 2.7 (for non-negative samples)
2. Check that the report MIC4109 only has swedish language, and that the language is sensible
3. Check that the typing report for MIC4109 looks normal in printing (apple-p) mode; e.g. no unreadable columns or bad table splits
4. Check that the Teknisk beskrivning av analysen, Analysbegränsningar för, Avvikelser från metoden, Signatur för godkännande av rapport are all present and sensible.
5. Check that the QC thresholds mentioned in the config are correctly appled to all QC reports.
6. Check that the database mentioned in the configuration has group write access.
7. Go to the folder mentioned under config['folders']['reports'] and check for duplicates of the reports generated
8. Pick a subset of samples of MIC4109 (at least 3). Doublecheck that  Ankomstdatum, Datum prep, datum sekvensering, Metod prep and Metod sekvensering mimic the values in the LIMS.
9. Check that Verifierad is correctly established for at least 5 samples.
10. Create a fake fastq file of size 2gb and run through microSALT analyse. Verify that it sends a warning. Immediately cancel the analysis afterwards.
11. Rename one of the samples in MIC4109 to its external name counterpart. Verify that the run still starts and resolves.

## Test results
The microSALT utils report command had to be utilized since currently hasta cannot send e-mails automatically.
1. MIC4107A6 now has a ST due to updated databases. Control response is slightly differently phrased. Otherwise; MIC3109 QC and MIC4107 & MIC4109 Typing and Resistance results are identical.
2. Language is all swedish and understandable. Only phrases "Datum prep" and "Datum sekvensering" sounds a bit stilted.
3. In printing mode for the Typing report of MIC4109 all tables are legible and without bad splits
4. All four sections are present in all reports and legible
5. Reports MIC4109 and MIC4107 have the QC highlights consistently applied.  However, MIC4107 had several samples hit bp_10x_fail threshold. Consider reducing it to 75-70%.
6. Database in /home/proj/stage/microbial/meta/ has group write access
7. Duplicate reports are found in /home/proj/stage/microbial/results/reports/
8. Checked MIC4109A1-A5. For all samples, 
ankomstdatum = CG002 - Reception Control Date arrived at.., 
libprep datum = last CG002 - Aggregate QC (Library Validation) date, 
sekvenserings datum = finish date i CG002 - Illumina Sequencing (Illumina ???), 
metod prep = CG002 - Microbial Library Prep (Nextera), 
metod sekvensering = CG002 - Cluster Generation Metod & Document
9. Verified has correctly been established for all samples in MIC4107. Sample A4 and A5 correctly lacks verification flag.
10. Created two fake files in fake project MIC3109A1 with the command `dd if=/dev/zero of=file.txt count=1024 bs=1048576`. The message triggers.
11. Sample MIC4109A1 was renamed to 18ET500035. Analysis started successfully.


# Sign-offs
